### PR TITLE
Builder etc fixes

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- send sounds to tracking pos to avoid the brrrrrrrr
- display homeless string in assignment ui
- handle builder water more smoothly
- handle placing blockstates better

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java [hunk 1]
```diff
@@ -1944,14 +1945,16 @@ public void refreshDimensions()
     @Override
     public void queueSound(@NotNull final SoundEvent soundEvent, final BlockPos pos, final int length, final int repetitions)
     {
-        Network.getNetwork().sendToTrackingEntity(new PlaySoundForCitizenMessage(this.getId(), soundEvent, this.getSoundSource(), pos, level, length, repetitions), this);
+        Network.getNetwork().sendToPosition(new PlaySoundForCitizenMessage(this.getId(), soundEvent, this.getSoundSource(), pos, level, length, repetitions),
+          new PacketDistributor.TargetPoint(pos.getX(), pos.getY(), pos.getZ(), BLOCK_BREAK_SOUND_RANGE, level.dimension()));
```
Comment (someaddons) [inline src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java:1949]: Not sure if this makes any difference? the previous sends it to all players who can see/are near the entity, the new one sends it to all players who are near the entity position

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java:1949]: I think entity tracking goes much much farther than pos tracking. Like we see entities much farther away.

Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java:1949]: nah pos tracking is further, its the chunks you see around you, entity is closer somewhat

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java:1949]: Using BLOCK_BREAK_SOUND_RANGE for 16 blocks range

Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/citizen/EntityCitizen.java:1949]: ye that is shorter then

**Review by someaddons — APPROVED** [review]

**Pull request merged**